File: Files/MVP.xctemplate/___FILEBASENAME___/___FILEBASENAME___Router.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//
 
#import "___VARIABLE_mvpModuleName___Router.h"
#import "___VARIABLE_mvpModuleName___Presenter.h"
#import "___VARIABLE_mvpModuleName___ViewController.h"

@implementation ___VARIABLE_mvpModuleName___Router

+ (UIViewController *)createModule
{
    NSString *viewName = NSStringFromClass([___VARIABLE_mvpModuleName___ViewController class]);
    ___VARIABLE_mvpModuleName___ViewController *viewController = [[___VARIABLE_mvpModuleName___ViewController alloc] initWithNibName:viewName bundle:nil];
    ___VARIABLE_mvpModuleName___Router *router = [[___VARIABLE_mvpModuleName___Router alloc] init];
    ___VARIABLE_mvpModuleName___Presenter *presenter = [[___VARIABLE_mvpModuleName___Presenter alloc] initWithInterface:viewController router:router];
    viewController.presenter = presenter;
    router.viewController = viewController;
    return viewController;
}
 
@end

File: Files/VPER.xctemplate/___FILEBASENAME___/___FILEBASENAME___Router.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created ___FULLUSERNAME___ on ___DATE___.
//

import UIKit

final class ___VARIABLE_productName:identifier___Router: ___VARIABLE_productName:identifier___WireframeProtocol {
    
    weak var view: UIViewController?
    
    static func createModule() -> UIViewController {
        let view = ___VARIABLE_productName:identifier___ViewController(nibName: nil, bundle: nil)
        let router = ___VARIABLE_productName:identifier___Router()
        let presenter = ___VARIABLE_productName:identifier___Presenter(interface: view, router: router)
        
        view.presenter = presenter
        router.view = view
        
        return view
    }
}
